catalog/src/main/kotlin/com/adevinta/spark/catalog/examples/samples/text/DropdownExamples.kt

```diff
@@ -74,15 +77,91 @@ public val DropdownsExamples: List<Example> = listOf(
+        sourceUrl = DropdownsExampleSourceUrl,
+    ) {
+        val multiSelectionFilter by remember {
+            mutableStateOf(DropdownStubData)
+        }
+
+        var selectedItems by remember {
+            mutableStateOf(
+                listOf(
+                    "To Kill a Mockingbird",
+                    "War and Peace",
+                ),
+            )
+        }
+        val multiSelectedValues by remember(selectedItems.size) {
+            derivedStateOf {
+                val suffix = if (selectedItems.size > 1) ", +${selectedItems.size - 1}" else ""
+                selectedItems.firstOrNull().orEmpty() + suffix
+            }
+        }
+        var expanded by remember { mutableStateOf(false) }
+        Dropdown(
+            modifier = Modifier.fillMaxWidth(),
+            value = multiSelectedValues,
+            label = "Book",
+            placeholder = "Pick a Book",
+            expanded = expanded,
+            onExpandedChange = {
+                expanded = !expanded
+            },
+            onDismissRequest = {
+                expanded = false
+            },
+            dropdownContent = {
+                multiSelectionFilter.forEach { group ->
+                    DropdownMenuGroupItem(
+                        title = {
+                            Text(group.name)
+                        },
+                    ) {
+                        group.books.forEach { book ->
+                            // This should be part of your model otherwise it's a huge work that done on
+                            // each items but we're simplifying things since it's an example here.
+                            val isSelected = book in selectedItems
+                            DropdownMenuItem(
+                                text = {
+                                    Text(
+                                        text = book,
+                                        style = if(isSelected) {
+                                            SparkTheme.typography.body1.highlight
+                                        } else {
+                                            SparkTheme.typography.body1
+                                        }
+                                    )
+                                },
+                                onClick = {
+                                    selectedItems = if (book in selectedItems) {
+                                        selectedItems - book
+                                    } else {
+                                        selectedItems + book
+                                    }
+                                    // Here we want to have the dropdown to stay expanded when we multiSelect but
+                                    // you could use the line below if in the case you want to have the same behaviour
+                                    // as the single selection.
+                                    // expanded = false
```

Comment:
I would suggest to extract this part, to make the example more concise, easy to go through the code